packages/app/src/cli/services/generate-schema.ts

```diff
@@ -0,0 +1,30 @@
+import {session, output, api, store} from '@shopify/cli-kit'
+import {AppInterface} from '../models/app/app.js'
+import {FunctionExtension} from '../models/app/extensions.js'
+import {getAppIdentifiers} from '../models/app/identifiers.js'
+import {fetchOrganizationAndFetchOrCreateApp} from './environment.js'
+
+interface GenerateSchemaOptions {
+  app: AppInterface
+  extension: FunctionExtension
+  apiKey?: string
+}
+
+export async function generateSchemaService(options: GenerateSchemaOptions) {
+  const {extension, app} = options
+  const token = await session.ensureAuthenticatedPartners()
+  const {apiVersion: version, type} = extension.configuration
+  const apiKey =
+    options.apiKey ||
+    getAppIdentifiers({app}).app ||
+    (await fetchOrganizationAndFetchOrCreateApp(app, token)).partnersApp.apiKey
```

Comment:
@pepicrft I'm not sure if having this fallback to prompt makes sense if the user is going to consume the output (e.g. pipe to code generation tool). What do you think?